api/utilities.py

```diff
@@ -137,3 +147,159 @@ async def add_parameter_name_and_datetime(request, parameter_name: str | None, d
         start, end = get_datetime_range(datetime)
         request.temporal_interval.start.CopyFrom(start)
         request.temporal_interval.end.CopyFrom(end)
+
+    if standard_names:
+        request.filter["standard_name"].values.extend(split_and_strip(standard_names))
+
+    if methods:
+        request.filter["function"].values.extend(split_and_strip(methods))
+
+    if periods:
+        request.filter["period"].values.extend(await get_periods_from_request(periods))
+
+
+async def get_periods_from_request(periods: str | None) -> list[str]:
+    split_on_slash = periods.split("/")
+    if len(split_on_slash) == 1:
+        return split_and_strip(periods)
+    elif len(split_on_slash) == 2:
+        return await get_iso_8601_range(split_on_slash[0].upper(), split_on_slash[1].upper())
+    else:
+        raise HTTPException(status_code=400, detail=f"Invalid ISO 8601 range format: {periods}")
+
+
+def get_z_levels_or_range(z: str | None) -> list[float]:
+    """
+    Function for getting the z values from the z parameter.
+    """
+    # it can be z=value1,value2,value3: z=2,10,80
+    # or z=minimum value/maximum value: z=10/100
+    # or z=Rn/min height/height interval: z=R20/100/50
+    try:
+        split_on_slash = z.split("/")
+        if len(split_on_slash) == 2:
+            z_min = float(split_on_slash[0]) if split_on_slash[0] != ".." else float("-inf")
+            z_max = float(split_on_slash[1]) if split_on_slash[1] != ".." else float("inf")
+            return [z_min, z_max]
+        elif len(split_on_slash) > 2:
+            return get_z_values_from_interval(split_on_slash)
+        else:
+            return list(map(float, z.split(",")))
+    except ValueError:
+        raise HTTPException(status_code=400, detail=f"Invalid levels value: {z}")
+
+
+def get_z_values_from_interval(interval: list[str]) -> list[float]:
+    """
+    Function for getting the z values from a repeating-interval pattern.
+    """
+    # Make sure the pattern is Rn/n/n
+    # Allow optional decimals in interval and starting value
+    # TODO: Can the starting value be negative? Can the interval be negative to go in reverse?
+    pattern = re.compile(r"^R\d+/\d+(\.\d+)?/\d+(\.\d+)?$")
+    if not pattern.match("/".join(interval)):
+        raise HTTPException(status_code=400, detail=f"Invalid levels repeating-interval: {'/'.join(interval)}")
+
+    amount_of_intervals = int(interval[0][1:])
+    interval_value = float(interval[1])
+    starting_value = float(interval[2])
```

Comment:
Not consistent with line 177. Not sure which is correct.

Reply:
Somewhere in the making they went into wrong order, but fixed in: https://github.com/EURODEO/e-soh/pull/146/commits/a9fcc8d3d1f51852b059081298641c22e9d69569